modules/microsoft.cdn/azure-frontdoor-cdn-profile/1.0/README.md

```diff
@@ -0,0 +1,174 @@
+3. Create ruleSets. For example, with ModifyResponseHeader, RouteConfigurationOverride (Cache Override)
+4. Create waf with Custom rules in Block Mode. (In this example, blocking all method except GET, OPTIONS and HEAD)
+5. Create waf with managed rules in Log Mode.
+6. Attach waf as security policy to endpoint
+7. Dynamically create custom domain and their association
+8. Attach AFD provided managed certificate for TLS. 
+9. Dynamically create Origin and Origin Group using array and their attachment with Routes, WAF policy etc.
+10. Create event namespace and hub
+11. Create Diagnostic Settings using eventHub for sending Azure FrontDoor CDN logs to event Hub.
+
+## Parameters
+
+| Name | Type | Required | Description |
+| :------------- | :----------: | :----------: | :------------- |
+| skuName | string | Yes | Name of Azure CDN SKU. One of `Premium_AzureFrontDoor` or `Standard_AzureFrontDoor` |
+| envName | string | Yes | Environment Name for CDN Profile. |
+| enableAfdEndpoint | bool | Yes | Enable or Disable a CDN Endpoint/Profile. |
+| enableWAFPolicy | bool | Yes | Enable or Disable a WAF Security Policy |
+| wafPolicyMode | string | Yes | Policy Mode for WAF. One of `Detection` or `Prevention` |
+| customDomains | array | Yes | Array of Custom Domains  |
+| origins | array | Yes | Array of Origin containing origin hostname, origin group name, path pattern to attach to origin and  enableState.  |
+| cdnProfileTags | object | Yes | Tags for CDN Profile  |
+| eventHubName | string | Yes | Event Hub to create for receiving CDN Logs. |
+| eventHubNamespace | string | Yes | Event Hub Namespace name. |
+| eventHubLocation | string | Yes | One of Valid Azure Region. |
+
+
+## Directory Structure
+
+```bash
+.
+├── README.md
+├── azuredeploy.parameters.json
+├── images
+│   └── deployment.png
+├── main.bicep
+├── metadata.json
+└── modules
+    ├── diagnosticsettings.bicep
+    ├── eventhub.bicep
+    ├── profile.bicep
+    ├── routes.bicep
+    ├── rulesets.bicep
+    └── waf.bicep
+```
+
+1. Directory `modules` contains base bicep files:
+   1. `diagnosticSettings.bicep`: create diagnostic settings to send Azure cdn access logs to event hub.
+   2. `eventhub.bicep`: create eventhub namespace and eventhub instance.
+   3. `profile.bicep`: invoke modules to create cdn profile, rule sets and diagnostic settings.
+   4. `routes.bicep`: create cdn routes for profile.
+   5. `rulesets.bicep`: create rule sets that are required by CDN Profile.
+   6. `waf.bicep`: create WAF with Managed and Custom rules that needs to be attached to CDN Profile as Security Policy.
+3. `main.bicep` provides an abstracted view to a user for creating CDN profile and waf attachment.
+
+## Deployment
+
+### Setting environment variable
+
+```bash
+export CDN_SUBS_ID="9xaxx10b-0xx2-xxxx-9xx2-d81a9xxxx921"
+export CDN_RESOURCE_GROUP_NAME="afd-cdn-foss-rg"
+export BICEP_FILE_NAME="main.bicep"
+
+```
+
+### Login and set subscription context
```

Comment:
remove the deployment steps since they must be the same for all samples in the repo

Reply:
Fixed. Removed the deployment steps.